app/src/ui/create-branch/index.tsx

```diff
@@ -10,10 +10,10 @@ import { Button } from '../lib/button'
 import { LinkButton } from '../lib/link-button'
 import { ButtonGroup } from '../lib/button-group'
 import { Dialog, DialogError, DialogContent, DialogFooter } from '../dialog'
-import { Octicon, OcticonSymbol } from '../octicons'
 import { VerticalSegmentedControl } from '../lib/vertical-segmented-control'
 import { TipState, IUnbornRepository, IDetachedHead, IValidBranch } from '../../models/tip'
 import { assertNever } from '../../lib/fatal-error'
+import { renderBranchNameWarning } from '../lib/branch-names';
```

Comment:
At the risk of bike shedding, let's name the file something more specifically about warnings. `branch-name-warnings` maybe?